### DISCUSSION TITLE
Coloring the margintoc

### DISCUSSION THREAD

**Discussion by AlexanderZeilmann** [fmarotta/kaobook#197] [Q&A] [answered]
How can I set the font color in the margin toc?

The code 
```latex
% !TEX program = xelatex

\documentclass{kaobook}

\addtokomafont{section}{\color{blue}}

\begin{document}

\color{red}

\setchapterpreamble[u]{\margintoc}
\chapter{Chapter}
\section{Section}
Text

\end{document}

```
produces the following result:
![Bildschirmfoto 2022-01-03 um 15 27 59](https://user-images.githubusercontent.com/901391/147942168-608f0668-0b86-4953-b039-a9c816d26e16.png)
The section number/name and page number are always black, the dots use the color of the `section`.
This looks a little odd. So how can I set the font color in the margintoc?

The documentation of the `marginnote` package suggests 
```latex
\renewcommand*{\marginfont}{\color{red}\sffamily}
```
but this does not change anything for me.

Related bug: #190

**Comment by AlexanderZeilmann** [comment, marked as answer]
To answer my own question:

We can change the color in the margintoc by setting the colors right before `\chapter` and resetting them after it:

```latex
\hypersetup{linkcolor=orange}            % Set the color for the section number/name and page number in the margintoc
\addtokomafont{section}{\color{orange}}  % Set the color for the dots in the margin toc in the margintoc
\chapter{Chapter}
\hypersetup{linkcolor=green}             % Reset the linkcolor for the normal text
\addtokomafont{section}{\color{blue}}    % Reset the section color for the normal text
```

Instead of modifying every `chapter`-call, [we can redefine `\chapter`](https://tex.stackexchange.com/a/73252).

**Reply by fmarotta** [reply]
I'll try to define a new "komafont" for the margintoc, so that one could just say `\setkomafont{margintoc}{<font specification>}`.
